### PR TITLE
Faucet UI Improvements

### PR DESCRIPTION
**Pull request by AtelyPham**
## Summary of changes
- Improve the Faucet UI

### Proposed area of change

- [ ] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [ ] `apps/webbsite`
- [x] `apps/faucet`
- [ ] `apps/tangle-website`
- [ ] `libs/webb-ui-components`

### Reference issue to close (if applicable)
- Closes https://github.com/webb-tools/webb-dapp/issues/1326
- Closes https://github.com/webb-tools/webb-dapp/issues/1360

### CODE REVIEW

File: apps/faucet/src/components/ProcessingModal.tsx [hunk 1]
```diff
@@ -107,6 +109,41 @@ const ProcessingModal = () => {
...
+        mintTokenRes.getErrorCode() === FaucetErrorCode.TOO_MANY_CLAIM_REQUESTS
+      ) {
+        const payload = mintTokenRes.getPayload();
+
+        if (payload && 'lastClaimedDate' in payload) {
+          const { lastClaimedDate, claimPeriod } =
+            payload as ErrorPayload[FaucetErrorCode.TOO_MANY_CLAIM_REQUESTS];
+
+          if (lastClaimedDate && claimPeriod) {
+            const timeLeft =
```
Comment (vutuanlinh2k2) [inline apps/faucet/src/components/ProcessingModal.tsx:128]: maybe you can create an utility function and put it inside the `utils` folder

File: apps/faucet/src/components/ProcessingModal.tsx [hunk 2]
```diff
@@ -107,6 +109,41 @@ const ProcessingModal = () => {
     }
   }, [mintTokenRes]);
 
+  const errorMessage = useMemo(() => {
+    if (!mintTokenRes) return '';
```
Comment (vutuanlinh2k2) [inline apps/faucet/src/components/ProcessingModal.tsx:113]: I see there are 4 nested if statement right here. Maybe with this approach the code is easier to read (suggested by chatgpt):

```
const getErrorMessage = (mintTokenRes) => {
  if (!mintTokenRes) return '';

  const defaultMessage = 'Oops, the transfer could not be completed.';

  if (!FaucetError.isFaucetError(mintTokenRes)) {
    return defaultMessage;
  }

  const errorCode = mintTokenRes.getErrorCode();

  if (errorCode !== FaucetErrorCode.TOO_MANY_CLAIM_REQUESTS) {
    return defaultMessage;
  }

  const payload = mintTokenRes.getPayload();

  if (!payload || !('lastClaimedDate' in payload)) {
    return defaultMessage;
  }

  const { lastClaimedDate, claimPeriod } = payload;
  const timeLeft = Date.now() - lastClaimedDate.getTime() + claimPeriod;
  const timeStr = formatTimeLeft(timeLeft);

  if (timeStr) {
    return `You have already claimed within the specified time period. Please wait ${timeStr} before making another claim.`;
  }

  return defaultMessage;
};
```

**Review by vutuanlinh2k2 — COMMENTED** [review]
Left some comments!

**Review by vutuanlinh2k2 — APPROVED** [review]
LGTM 😊

**Review by monaiuu — COMMENTED** [review]
Looks amazing! Two tiny things from me:

Screenshot 2023-07-24 at 10 05 23 AM

- Let's keep all webb tokens naming convention consistent by using 'webbXYZ' rather than 'WebbXYZ'.

Screenshot 2023-07-24 at 10 05 45 AM

- The link button texts is using `body04` I believe, let's make it `body02` for improved readability and alignment. 

Also, would it be possible to review the end-to-end flow through the twitter login flow? I get an error currently.

**Comment by monaiuu** [comment]
LGTM, let's merge it. 💪

**Pull request merged**